librz/core/disasm.c

```diff
@@ -5785,28 +5789,26 @@ RZ_API int rz_core_print_disasm_instructions_with_buf(RzCore *core, ut64 address
 			ds->hint = NULL;
 		}
 	}
-	if (buf == core->block && nb_opcodes > 0 && j < nb_opcodes) {
-		rz_core_seek(core, core->offset + i, true);
-		goto toro;
-	}
 	rz_cons_break_pop();
 	ds_free(ds);
-	core->offset = old_offset;
 	rz_reg_arena_pop(core->analysis->reg);
-
+	if (alloc_buf) {
+		free(buf);
+	}
 	return len;
 }
 
 /**
- * \brief Converting negative numbers n_opcodes and n_opcodes
+ * \brief (DEPRECATED, consider rz_core_backward_offset)
+ *        Converting negative numbers n_opcodes and n_opcodes
  * 	  to positive numbers n_opcodes and n_opcodes
  *	  and seek the appropriate offset
  * \param core RzCore reference
  * \param pn_opcodes Pointer to n_opcodes
  * \param pn_bytes Pointer to n_bytes
  * \return success
  */
-RZ_IPI bool rz_core_handle_backwards_disasm(RZ_NONNULL RzCore *core,
+RZ_DEPRECATE RZ_IPI bool rz_core_handle_backwards_disasm(RZ_NONNULL RzCore *core,
```

Comment:
Many functions relying on core->block need to be changed to totally remove `rz_core_handle_backwards_disasm`. It will be a huge change considering the length of the function `rz_core_disasm_pdi_with_buf` and `rz_core_print_disasm`, etc. I think it may be better to do this in subsequent pr.